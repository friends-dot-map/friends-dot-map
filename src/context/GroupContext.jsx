import { createContext, useContext, useState, useEffect } from 'react';
import { getAllProfiles } from '../services/profiles';

export const GroupContext = createContext();

const GroupProvider = ({ children }) => {
  const [group, setGroup] = useState([]);

  useEffect(() => {
    const fetchGroup = async () => {
      try {
        const data = await getAllProfiles();
        setGroup(data);
      } catch (error) {
        throw error('something went horribly wrong');
      }
    };
    fetchGroup();
  }, []);
  const groupValue = { group, setGroup };
  return (
    <GroupContext.Provider value={groupValue}>{children}</GroupContext.Provider>
  );
};

const useGroup = () => {
  const context = useContext(GroupContext);

  if (context === undefined) {
    throw new Error('useGroup must be used within a GroupProvider');
  }

  return context;
};

export { GroupProvider, useGroup };
