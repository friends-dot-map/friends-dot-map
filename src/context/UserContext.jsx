import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
