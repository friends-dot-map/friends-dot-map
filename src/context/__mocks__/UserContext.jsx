import { createContext, useContext, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
