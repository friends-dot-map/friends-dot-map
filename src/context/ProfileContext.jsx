import { createContext, useContext, useState, useEffect } from 'react';
import { getProfile } from '../services/profiles';
import { useUser } from './UserContext';
import { formatDate } from '../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [loading, setLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = await getProfile(user.email);
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    loading,
    setLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
