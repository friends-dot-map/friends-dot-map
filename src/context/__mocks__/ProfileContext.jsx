import { createContext, useContext, useState, useEffect } from 'react';
import { useUser } from '../UserContext';
import { formatDate } from '../../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ mockProfile, children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [profileLoading, setProfileLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = mockProfile || {};
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setProfileLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    profileLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
