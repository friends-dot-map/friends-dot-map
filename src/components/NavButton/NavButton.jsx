import { Menu, Transition } from '@headlessui/react';
import { MenuIcon } from '@heroicons/react/solid';
import { Link } from 'react-router-dom';
import { Fragment } from 'react';
import { useUser } from '../../context/UserContext';
import { useProfile } from '../../context/ProfileContext';
import { signOutUser } from '../../services/users';

function classNames(...classes) {
  return classes.filter(Boolean).join(' ');
}

export default function NavButton({ activeOnMap = false }) {
  const { setUser } = useUser();
  const {
    setProfile,
    profile: { username },
  } = useProfile();

  let position, bgColor, size;

  if (activeOnMap) {
    position = 'absolute right-0 top-1/4';
    bgColor = 'bg-white ring-tint';
    size = 'h-6 w-6 ';
  } else {
    position = 'relative';
    bgColor = 'bg-white/0';
    size = 'h-10 w-10 md:h-16 md:w-16';
  }

  return (
    <Menu as="div" className={`${position} inline-block text-left`}>
      <div>
        <Menu.Button
          className={`${bgColor} inline-flex justify-center w-full rounded-md shadow-sm px-4 py-2 text-base font-medium text-dark hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500`}
        >
          <MenuIcon className={`${size}`} aria-hidden="true" />
        </Menu.Button>
      </div>
      <Transition
        as={Fragment}
        enter="transition ease-out duration-100"
        enterFrom="transform opacity-0 scale-95"
        enterTo="transform opacity-100 scale-100"
        leave="transition ease-in duration-75"
        leaveFrom="transform opacity-100 scale-100"
        leaveTo="transform opacity-0 scale-95"
      >
        <Menu.Items className="origin-top-right absolute right-0 mt-2 w-56 rounded-md shadow-lg bg-white ring-tint focus:outline-none">
          <div className="py-1">
            <Menu.Item>
              {({ active }) => (
                <Link
                  to="/"
                  className={classNames(
                    active ? 'bg-dark/10 text-dark' : 'text-dark/70',
                    'block px-4 py-2 text-lg'
                  )}
                >
                  Home
                </Link>
              )}
            </Menu.Item>
            <Menu.Item>
              {({ active }) => (
                <Link
                  to={`/profile/${username}`}
                  className={classNames(
                    active ? 'bg-dark/10 text-dark' : 'text-dark/70',
                    'block px-4 py-2 text-lg'
                  )}
                >
                  Your Profile
                </Link>
              )}
            </Menu.Item>
            <Menu.Item>
              {({ active }) => (
                <Link
                  to="/group"
                  className={classNames(
                    active ? 'bg-dark/10 text-dark' : 'text-dark/70',
                    'block px-4 py-2 text-lg'
                  )}
                >
                  Your Groups
                </Link>
              )}
            </Menu.Item>

            <Menu.Item>
              {({ active }) => (
                <button
                  onClick={async () => {
                    await signOutUser();
                    setUser({});
                    setProfile({});
                  }}
                  type="submit"
                  className={classNames(
                    active ? 'bg-dark/10 text-dark' : 'text-dark/70',
                    'block w-full text-left px-4 py-2 text-lg'
                  )}
                >
                  Log out
                </button>
              )}
            </Menu.Item>
          </div>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}
