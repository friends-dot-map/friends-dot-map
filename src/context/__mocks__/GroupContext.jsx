import { createContext, useContext, useState, useEffect } from 'react';
import { useProfile } from '../ProfileContext'


export const GroupContext = createContext();

const GroupProvider = ({ mockGroup, children }) => {
  const [group, setGroup] = useState([]);
  const { profile } = useProfile();

  useEffect(() => {
    const fetchGroup = async () => {
      try {
        const data = mockGroup || [];
        setGroup(data);
      } catch (error) {
        throw error('something went horribly wrong');
      }
    };
    fetchGroup();
  }, [profile]);

  const groupValue = { group };

  return (
    <GroupContext.Provider value={groupValue}>{children}</GroupContext.Provider>
  );
};

const useGroup = () => {
  const context = useContext(GroupContext);

  if (context === undefined) {
    throw new Error('useGroup must be used within a GroupProvider');
  }

  return context;
};

export { GroupProvider, useGroup };
