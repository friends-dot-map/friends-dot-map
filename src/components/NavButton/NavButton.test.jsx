import NavButton from "./NavButton";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom/";
import { ProfileProvider } from "../../context/ProfileContext";
import { GroupProvider } from "../../context/GroupContext";
import { UserProvider } from "../../context/UserContext";

test('auth form renders login page', () => {
    const { container } = render (
    <UserProvider>
        <ProfileProvider>
            <GroupProvider>
                <MemoryRouter >
                    <NavButton  />
                </MemoryRouter>
            </GroupProvider>
        </ProfileProvider>
    </UserProvider>
    
    );
    

    expect(container).toMatchSnapshot();
});