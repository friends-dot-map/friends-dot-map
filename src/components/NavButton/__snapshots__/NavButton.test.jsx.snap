// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth form renders login page 1`] = `
<div>
  <div
    class="relative inline-block text-left"
  >
    <div>
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="bg-white/0 inline-flex justify-center w-full rounded-md shadow-sm px-4 py-2 text-base font-medium text-dark hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500"
        id="headlessui-menu-button-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="h-10 w-10 md:h-16 md:w-16"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
